apps/coordinator/ts/scheduler/scheduler.service.ts

```diff
@@ -0,0 +1,292 @@
+      this.logger.error(`Error: ${ErrorCodes.POLL_ALREADY_TALLIED}, poll has already been tallied`);
+      throw new Error(ErrorCodes.POLL_ALREADY_TALLIED.toString());
+    }
+
+    const { isScheduled } = await this.isPollScheduled({
+      maciAddress,
+      pollId,
+      chain,
+    });
+
+    if (isScheduled) {
+      this.logger.error(`Error: ${ErrorCodes.POLL_ALREADY_SCHEDULED}, poll is already scheduled`);
+      throw new Error(ErrorCodes.POLL_ALREADY_SCHEDULED.toString());
+    }
+
+    await this.setupPollFinalization({
+      maciAddress,
+      pollId,
+      chain,
+      deploymentBlockNumber,
+      mode,
+      endDate,
+      merged: false,
+      proofsGenerated: false,
+    });
+
+    return { isScheduled: true };
+  }
+
+  /**
+   * Delete scheduled poll
+   * @param args - Arguments to identify the scheduled poll
+   * @returns Response object with isScheduled property indicating the poll is no longer scheduled
+   */
+  async deleteScheduledPoll(scheduledPoll: IIdentityScheduledPoll): Promise<IIsPollScheduledResponse> {
+    const key = getPollKeyFromObject(scheduledPoll);
+
+    try {
+      this.schedulerRegistry.deleteTimeout(key);
+    } catch (error) {
+      this.logger.error(`Failed to delete timeout for ${key}:`);
+    }
+
+    await this.redisService.delete(key);
+
+    return { isScheduled: false };
+  }
+
+  /**
+   * Restore all scheduled timeouts from Redis
+   */
+  private async restoreTimeouts(): Promise<void> {
+    const pollKeys = await this.redisService.getAll();
+
+    await Promise.all(
+      pollKeys.map(async (key) => {
+        this.logger.log(`Restoring timeout for poll key: ${key}`);
+        try {
+          const value = await this.redisService.get(key);
+
+          if (!value) {
+            await this.redisService.delete(key);
+            return;
+          }
+
+          const poll = JSON.parse(value) as IScheduledPoll;
+
+          const { endDate, isPollTallied } = await this.getPollFinalizationData({
+            maciAddress: poll.maciAddress,
+            pollId: poll.pollId,
+            chain: poll.chain,
+          });
+
+          if (isPollTallied) {
+            await this.deleteScheduledPoll({
+              maciAddress: poll.maciAddress,
+              pollId: poll.pollId,
+              chain: poll.chain,
+            });
+            return;
+          }
+
+          const { delay } = await this.setupPollFinalization({
+            maciAddress: poll.maciAddress,
+            pollId: poll.pollId,
+            mode: poll.mode,
+            deploymentBlockNumber: poll.deploymentBlockNumber,
+            chain: poll.chain,
+            merged: poll.merged,
+            proofsGenerated: poll.proofsGenerated,
+            endDate,
+          });
+
+          this.logger.log(`poll ${value} to be executed at ${new Date(Date.now() + delay).toISOString()}`);
+        } catch (error) {
+          this.logger.error(`Error restoring timeout for poll key: ${key}`, error);
+        }
```

Comment:
Why is this error ignored?

Reply:
If  one poll cannot be restored we are logging the error but not stopping the process of restoring the other ones